File: Practice/24/C++/Project/Project/Project.cpp
#include <iostream>
#include <json.hpp>
#include <fstream>
#include <string>
#include <map>

using json = nlohmann::json;
int main()
{
    std::ifstream File("in.json");
    nlohmann::json Jstr;
    nlohmann::json Jstr2;
    File >> Jstr;

    std::map<int, int> Buf;
    int UserID;
    bool TaskComp;
  
    for (int i = 0; i < Jstr.size(); i++) {

        UserID = Jstr[i]["userId"];
        int bufId = UserID;

        TaskComp = Jstr[i]["completed"];

        if (TaskComp == true) {
           Buf[UserID] += 1;
        }
    }

    for (auto[userId, complete]: Buf) {
        Jstr2.push_back({ {"userId", userId},  {"task_completed", complete } });

    }

    std::ofstream OutFile("out.json");
    OutFile << Jstr2 << std::endl;
}

File: Practice/24/Python/Project.py
import json
with open("in.json", "r") as read_file:
    Jstr = json.load(read_file)
Buf = []
Res = []
i = 0
while i < len(Jstr):
    UserID = Jstr[i]["userId"]
    TaskComp = Jstr[i]["completed"]
    if TaskComp:
        Buf.append(UserID)
    i += 1
unique = list(set(Buf))
j, n = 0, 0
while j < len(unique):
    Res.append({ 'task_completed': Buf.count(unique[j]) , 'userId': unique[j]})
    j += 1
with open("out.json", "w") as write_file:
    json.dump(Res, write_file, indent=2)